### PR TITLE
LR-241 - UserOrg Identity manager should be optional for setup

### PR DESCRIPTION
**Pull request by BharathwajShankar**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: controller/conf/application.conf [hunk 1]
```diff
@@ -742,6 +742,9 @@ play.cache {
   #bindCaches = ["db-cache", "user-cache", "session-cache"]
 }
 
+#Keycloak optional config
```
Comment (reshmi-nair) [inline controller/conf/application.conf:745]: Remove Keycloak word from this and other comments and varibles. We can mention it as Auth Enabled

File: controller/app/util/RequestInterceptor.java [hunk 2]
```diff
@@ -43,46 +45,6 @@ private RequestInterceptor() {}
 
     // ---------------------------
     short var = 1;
-    apiHeaderIgnoreMap.put("/v1/user/create", var);
```
Comment (reshmi-nair) [inline controller/app/util/RequestInterceptor.java:46]: @BharathwajShankar Why have we remove all these lines? These doesnot require user auth token in any env. so pls add these back

File: core/platform-common/src/main/java/org/sunbird/keys/JsonKey.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
 public final class JsonKey {
   public static final String ANONYMOUS = "Anonymous";
   public static final String UNAUTHORIZED = "Unauthorized";
+  public static final String KEY_CLOAK_AUTH_ENABLED = "keycloakAuthenticationEnabled";
```
Comment (reshmi-nair) [inline core/platform-common/src/main/java/org/sunbird/keys/JsonKey.java:14]: Remove keycloak word

**Review by reshmi-nair — CHANGES_REQUESTED** [review]

File: core/platform-common/src/main/java/org/sunbird/keys/JsonKey.java (hunk 3 above)
Comment (reshmi-nair) [reply core/platform-common/src/main/java/org/sunbird/keys/JsonKey.java]: resolved

File: controller/conf/application.conf (hunk 1 above)
Comment (reshmi-nair) [reply controller/conf/application.conf]: resolved

File: controller/app/util/RequestInterceptor.java (hunk 2 above)
Comment (BharathwajShankar) [reply controller/app/util/RequestInterceptor.java]: added back the lines.

**Review by reshmi-nair — APPROVED** [review]

**Pull request merged**